File: 2010_2011_mf_sujet_6_ks_regression_lineaire/DKS2.py
import numpy as np
from scipy.stats import kstwobign

from frep2 import frep2


def DKS2(X, Y, alpha):
    """Two-sample Kolmogorov-Smirnov test. Returns p-value and decision."""
    n, m = len(X), len(Y)
    Z = np.sort(np.concatenate([X, Y]))
    T = np.array([abs(frep2(X, z) - frep2(Y, z)) for z in Z])
    d = np.sqrt(n * m / (n + m)) * np.max(T)
    print(f'd = {d}')
    ac = 1 - kstwobign.cdf(d)
    rep = 'OUI' if ac > alpha else 'NON'
    print(rep)
    return ac, rep

File: 2010_2011_mf_sujet_6_ks_regression_lineaire/DKS2.m
function [ac,rep ] = DKS2(X,Y,alpha )
%UNTITLED3 Summary of this function goes here
%   Detailed explanation goes here
Z=[X,Y];
n=length(X);
m=length(Y);
T=zeros(1,n+m);
Z=sort(Z);
for k=1:(n+m)
    T(k)=abs(frep2(X,Z(k))-frep2(Y,Z(k)));
end
d=sqrt(n*m/(n+m))*max(T)
ac=1-pks(d);
if (ac>alpha) rep = 'OUI'
else rep = 'NON'
end
end

